File: 프로그래머스/1/176963. 추억 점수/추억 점수.py
def solution(name, yearning, photo):
    answer = []

    pair = dict(zip(name, yearning))

    for i in range(len(photo)):
        answer1 = []
        for k in range(len(photo[i])):
            for j in range(len(name)):
                if (photo[i][k] == name[j]):
                    answer1.append(pair.get(name[j]))
        a = sum(answer1)
        answer.append(a)
        print(a)

    return answer

File: 프로그래머스/1/176963. 추억 점수/추억 점수.java
import java.util.HashMap;
import java.util.Map;
import java.util.Scanner;

class Solution {
    public int[] solution(String[] name, int[] yearning, String[][] photo) {
        Map<String, Integer> dic = new HashMap<String, Integer>();
        int n = photo.length;
        int[] answer = new int[n];
        for(int i=0;i< name.length;i++)
        {
            dic.put(name[i],yearning[i]);
        }
        int i=0;

        for(String[] row: photo)
        {
            int score = 0;

            for(String key: row)
            {
                if(dic.containsKey(key))
                {
                    score += dic.get(key);
                }
                else
                {
                    ;
                }
            }
            answer[i] = score;
            i++;
        }

        return answer;
    }
}